legal-api/src/legal_api/services/authz.py

```diff
@@ -669,3 +671,61 @@ def add_allowable_filing_type(is_allowable: bool = False,
         allowable_filing_types.append(allowable_filing_type)
 
     return allowable_filing_types
+
+
+def are_digital_credentials_allowed(business: Business, jwt: JwtManager):
+    """Return True if the business is allowed to have/view a digital business card."""
+    if not (token := pyjwt.decode(jwt.get_token_auth_header(), options={'verify_signature': False})):
+        return False
+
+    if not (user := User.find_by_jwt_token(token)):
+        return False
+
+    is_staff = jwt.contains_role([STAFF_ROLE])
+
+    is_sole_prop = business and business.legal_type == Business.LegalTypes.SOLE_PROP.value
+
+    is_login_source_bcsc = user.login_source == 'BCSC'
+
+    is_owner_operator = is_self_registered_owner_operator(business, user)
+
+    return is_login_source_bcsc and is_sole_prop and is_owner_operator and not is_staff
+
+
+def is_self_registered_owner_operator(business, user):
+    """Return True if the user is the owner operator of the business."""
+    if not (registration_filing := get_registration_filing(business)):
+        return False
+
+    if not (filing_json := registration_filing.filing_json.get('filing', {}).get('registration', None)):
+        return False
+
+    if not (parties := filing_json.get('parties', None)):
+        return False
+
+    completing_party = next((party for party in parties if party.get('roles', None) and
+                             any(role for role in party.get('roles')
+                                 if role.get('roleType') == 'completingParty')), None)
+    proprietor = next((party for party in parties if party.get('roles', None) and
+                       any(role for role in party.get('roles')
+                           if role.get('roleType') == 'proprietor')), None)
+    if (not completing_party) or (not proprietor):
+        return False
+
+    return completing_party.get('firstName') == proprietor.get('firstName') and \
+        completing_party.get('lastName') == proprietor.get('lastName') and \
+        proprietor.get('firstName') == user.firstname and \
+        proprietor.get('lastName') == user.lastname
```

Comment:
I think we should also grab the user that was associated with the registration filing and maybe compare the `idp_user_id`

Reply:
That's the id that uniquely identifies the user object.  

In this case it would just verify that the current logged in user is the same user that created the registration filing.